subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerExecutorLegacyApiIntegrationTest.groovy

```diff
@@ -186,4 +275,40 @@ class WorkerExecutorLegacyApiIntegrationTest extends AbstractIntegrationSpec {
             }
         """
     }
+
+    String getOptionVerifyingRunnable() {
+        return """
+            import java.util.regex.Pattern;
+            import java.util.List;
+            import java.lang.management.ManagementFactory;
+            import java.lang.management.RuntimeMXBean;
+            import javax.inject.Inject;
+
+            public class OptionVerifyingRunnable extends TestRunnable {
+                @Inject
+                public OptionVerifyingRunnable(String text, String[] arrayOfThings, ListProperty<String> listOfThings) {
+                    super(text, arrayOfThings, listOfThings);
+                }
+
+                public void run() {
+                    RuntimeMXBean runtimeMxBean = ManagementFactory.getRuntimeMXBean();
+                    List<String> arguments = runtimeMxBean.getInputArguments();
+                    assert arguments.contains("-Dfoo=bar");
+                    assert arguments.contains("-Dbar=baz");
+                    assert arguments.contains("-Xmx128m");
+                    assert arguments.contains("-Xms128m");
+                    assert arguments.contains("-Dfile.encoding=UTF-8");
+                    assert arguments.contains("-ea");
```

Comment:
❓ Nit: Would there be a way to centralize those configurations so it's declared once but used in multiple places?

Reply:
Yeah, this has always bothered me.  I made it a little tighter by using a "verifier" object that produces the dsl to configure it, captures the process options as a json file, and then verifies the options match.

Reply:
See https://github.com/gradle/gradle/pull/10047/commits/3d06528e6994b508a0300f0b873f6da89496fa14